maintenance/GenerateMissingIdentifiers.php

```diff
@@ -20,7 +21,24 @@ public function __construct() {
 	}
 
 	public function execute(): void {
-		$this->output( 'TODO' );
+		$pageIds = $this->getPageIds();
+		$pageIdsCount = count( $pageIds );
+		$creator = PersistentPageIdentifiersExtension::getInstance()->newCreatePersistentPageIdentifier();
+
+		foreach ( $pageIds as $pageId ) {
+			$creator->createId( intval( $pageId ) );
+		}
+
+		$this->output( "Created $pageIdsCount persistent IDs\n" );
+	}
+
+	private function getPageIds(): array {
+		return PersistentPageIdentifiersExtension::getInstance()->getDatabase()->newSelectQueryBuilder()
+			->select( 'p.page_id' )
+			->from( 'page', 'p' )
+			->leftJoin( 'persistent_page_ids', 'ppi', 'p.page_id = ppi.page_id' )
+			->where( 'ppi.page_id IS NULL' )
+			->fetchFieldValues();
```

Comment:
This works for all non-deleted pages.
If we need to generate persistent IDs for deleted pages then we'll need to do a similar query, but using field `ar_page_id` on the `archive` table too.